pageserver/src/layered_repository.rs

```diff
@@ -1359,6 +1392,14 @@ impl LayeredTimeline {
             layer.unload()?;
         }
 
+        drop(layers);
```

Comment:
I don't see any obvious races that could be caused by this, but shout out if one comes to mind